drkafka/src/test/java/com/pinterest/doctorkafka/KafkaClusterTest.java

```diff
@@ -119,9 +140,103 @@ void getAlternativeBrokersDuplicateReassignmentTest() throws Exception{
     assertNotEquals(altBrokers.get(1), altBrokers.get(2));
     // The broker queue should contain the same amount of brokers
     assertEquals(beforeSize, brokerQueue.size());
+  }
 
-    ReplicaStatsManager.bytesInStats = oldBytesInStats;
-    ReplicaStatsManager.bytesOutStats = oldBytesOutStats;
+  @Test
+  void testLocalityAwareReassignments() throws Exception {
```

Comment:
could we add another test to test the scenario that there is not enough capacity in the rack for assigning the partition?